### PR TITLE
Give Go module initialized with `dispatch init` the same name as it's directory name

### PR DESCRIPTION
**Pull request by chicoxyzzy**
This code patches the first line of the `go.mod` file to set a module name to a directory name.

### CODE REVIEW

File: cli/init.go [hunk 1]
```diff
@@ -256,6 +257,46 @@ func copyFile(srcFile string, dstFile string) error {
...
+	// Join the lines back into a single string with newlines
+	output := strings.Join(lines, "\n") + "\n"
+
+	// Write the modified content back to the file
+	err = os.WriteFile(goModPath, []byte(output), 0644)
+	if err != nil {
+		return err
+	}
+
+	// TODO: create .gitignore file?
```
Comment (achille-roussel) [inline cli/init.go:295]: Could we add a `.gitignore` file to the template so we don't need this to be a special case?

**Review by achille-roussel — APPROVED** [review]
Nice improvement 👍

File: cli/init.go (hunk 1 above)
Comment (chicoxyzzy) [reply cli/init.go:295]: I initially added it, but we never know what will be the `-o` flag value. I made it equal to the directory name as well initially, but that's also dynamical.

What known Go directories and files I should exclude?

Comment (chicoxyzzy) [reply cli/init.go:295]: What do you think about these two for Go and Python?

https://github.com/github/gitignore/blob/main/Go.gitignore
https://github.com/github/gitignore/blob/main/Python.gitignore

Comment (achille-roussel) [reply cli/init.go:295]: Let's roll with these 👍

**Comment by chicoxyzzy** [comment]
Merging this. See #82 for `.gitignore`

**Pull request merged**